platform/platform-crypto-provider/src/main/java/org/codice/ddf/platform/crypto/provider/MerlinExtension.java

```diff
@@ -0,0 +1,340 @@
+            }
+
+            PKIXParameters param = createPKIXParameters(set, enableRevocation);
+
+            // Generate cert path
+            if (foundIssuingCertChains != null && !foundIssuingCertChains.isEmpty()) {
+                java.security.cert.CertPathValidatorException validatorException = null;
+                // Try each potential issuing cert path for a match
+                for (Certificate[] foundCertChain : foundIssuingCertChains) {
+                    X509Certificate[] x509certs = new X509Certificate[foundCertChain.length + 1];
+                    x509certs[0] = certs[0];
+                    System.arraycopy(foundCertChain, 0, x509certs, 1, foundCertChain.length);
+
+                    List<X509Certificate> certList = Arrays.asList(certs);
+                    CertPath path = getCertificateFactory().generateCertPath(certList);
+
+                    try {
+                        validator.validate(path, param);
+                        // We have a valid cert path at this point so break
+                        validatorException = null;
+                        break;
+                    } catch (java.security.cert.CertPathValidatorException e) {
+                        validatorException = e;
+                    }
+                }
+
+                if (validatorException != null) {
+                    throw validatorException;
+                }
+            } else {
+                List<X509Certificate> certList = Arrays.asList(certs);
+                CertPath path = getCertificateFactory().generateCertPath(certList);
+
+                validator.validate(path, param);
+            }
+        } catch (NoSuchProviderException | NoSuchAlgorithmException
+                | CertificateException | InvalidAlgorithmParameterException
+                | java.security.cert.CertPathValidatorException
+                | KeyStoreException e) {
+            throw new WSSecurityException(
+                    WSSecurityException.ErrorCode.FAILURE, e, "certpath"
+            );
+        }
+
+        // Finally check Cert Constraints
+        if (!matchesSubjectDnPattern(certs[0], subjectCertConstraints)) {
+            throw new WSSecurityException(WSSecurityException.ErrorCode.FAILED_AUTHENTICATION);
+        }
+    }
+
+    private void buildTrustAnchors(Set<TrustAnchor> trustAnchorSet,
+            Enumeration<String> truststoreAliases) throws KeyStoreException {
+        while (truststoreAliases.hasMoreElements()) {
+            String alias = truststoreAliases.nextElement();
+            X509Certificate cert = (X509Certificate) truststore.getCertificate(alias);
+            if (cert != null) {
+                byte[] nameConstraints = null;
+                try {
+                    nameConstraints = extractNameConstraints(cert);
+                } catch (IOException e) {
+                    LOG.debug("Unable to decode name constraints.", e);
```

Comment:
In general, the _right_ reason to check sometimes and not others is to avoid wasting time when interpolated values require processing. In the case of this particular class, most of the logging is in the original version from WSS4J that has been modified so it's possible some unnecessary log level checks were occurring.